### PR TITLE
Documentation update for 1.0 release

### LINKED ISSUE #80: Prepare end-user documentation
NOTE: @tadeboro will (ab)use this issue for storing notes about the documentation. You have been warned.

Sections that we need to prepare are:

  * [x] quickstart,
  * [x] installation,
  * [x] a few scenario guides (simple installation, cluster setup),
  * [x] module API documentation (create a nicer layout),
  * [x] development guides (partially done).

### PR DESCRIPTION
**Pull request by tadeboro**
This PR contains changes that prepare the documentation for 1.0 release. Details are in the commit messages.

FIxes #80

### CODE REVIEW

File: docs/source/roles/backend.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
+Sensu Go backend role
+=====================
+
+This role installs, configures and starts the Sensu Backend service.
```
Comment (mancabizjak) [inline docs/source/roles/backend.rst]: Maybe we should use `sensu-backend`/_sensu-backend_ instead of Sensu Backend?

File: docs/source/roles/agent.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+Sensu Go agent role
+===================
+
+This role installs, configures and starts the Sensu Agent service.
```
Comment (mancabizjak) [inline docs/source/roles/agent.rst]: Maybe we should use `sensu-agent`/_sensu-agent_ instead of Sensu Agent?

File: docs/source/roles/backend.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+and configure it. We can customize the backend's configuration by adding more
+options to the *backend_config* variable.
+
+
+Backend configuration options
+-----------------------------
+
+The *backend_config* variable can contain any option that is valid for the
+Sensu Go backend version we are installing. All valid options are listed in
+the `official Sensu documentation`_.
```
Comment (mancabizjak) [inline docs/source/roles/backend.rst]: Same comment as for the agent role - I think it would be beneficial to be more explicit about 1:1 mapping of configuration options. I mean, we kind of already are specific in saying anything that's valid for the `sensu-backend` is also valid here. But I'd like to be very specific about the naming notation.

On second thought, this is already illustrated in the examples for both roles.
Feel free to do as you please :).

File: docs/source/roles/backend.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+   * - api_trusted_ca_file
+     - files/pki/sensu-api-ca.crt
+     - Path to the trusted certificate authority for the Sensu Go API
+       certificates.
+
+Dashboard
+^^^^^^^^^
+
+To secure the dashboard communication, create the appropriate files for the
```
Comment (mancabizjak) [inline docs/source/roles/backend.rst]:
```suggestion
To secure the Sensu dashboard communication, create the appropriate files for the
```

File: docs/source/roles/install.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+   This role only configures your package manager (like yum or apt) and
+   installs the binaries. It does **not** configure anything and it does
+   **not** run any services.
+
+
+Example playbook
+----------------
+
+The next :download:`playbook <../../examples/roles/install.yaml>` demonstrates
+how we can write most common tasks.
```
Comment (mancabizjak) [inline docs/source/roles/install.rst]:
```suggestion
how to install latest versions of _sensu-agent_ and _sensu-backend_.
```

File: docs/source/roles/install.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+   * - components
+     - - sensu-go-backend
+       - sensu-go-agent
+       - sensu-go-cli
+     - List of components to install. Valid values are ``sensu-go-backend``,
+       ``sensu-go-agent`` and ``sensu-go-cli``.
+
+   * - channel
+     - stable
+     - Repository channel that serves as a source of packages. We can see all
```
Comment (mancabizjak) [inline docs/source/roles/install.rst]:
```suggestion
     - Repository channel that serves as a source of packages. We can see all the
```

File: docs/examples/roles/install.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  vars:
+    components: [ sensu-go-cli ]
+
+- name: Install sensu-agent binary from testing repository
+  hosts: agents
+  roles:
+     - sensu.sensu_go.install
+  vars:
+    components: [ sensu-go-agent ]
+    channel: testing
```
Comment (mancabizjak) [inline docs/examples/roles/install.yaml]: Would you mind adding a task that installs `sensuctl` to localhost? Otherwise I think the fact that the role allows us to install `sensuctl` kind of gets lost.

File: docs/source/modules.rst [hunk 8]
```diff
@@ -0,0 +1,146 @@
+Modules
+=======
+
+While different modules perform different jobs, their interfaces all follow
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
While different modules perform different tasks, their interfaces all follow
```

File: docs/source/modules.rst [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+the same pattern as much as possible.
+
+The API of each module is composed of two parts. The *auth* parameter contains
+the pieces of information that are related to the Sensu Go backend that the
+module is connecting to. All other parameters hold the information related to
+the resource that we are operating on. Typical task would look like this:
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
the resource that we are operating on. A typical task would look like this:
```

File: docs/source/modules.rst [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+     register: result
+
+Info modules always return a list of objects that Sensu API returned. And if
+we try to fetch a non-existing resource, the result will hold an empty list.
+This makes it easy to write a conditional tasks using next pattern:
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
This makes it easy to write conditional tasks using next pattern:
```

File: docs/source/modules.rst [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+     register: result
+
+   - name: Do something if asset is there
+     debug:
+       msg: We are doing somethin
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
       msg: We are doing something
```

File: docs/source/modules.rst [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+       builds: ...
+
+Almost every module for manipulating resources has its info counterpart that
+can be used to retrieve information about the certain resources. For example,
+to get a list of assets from the ``my-namespace`` namespace, we would write a
+task like this one:
+
+.. code-block:: yaml
+
+   - name: Fetch list of all assets
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
   - name: Fetch a list of all assets
```

File: docs/source/modules.rst [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+   - name: Do something if asset is there
+     debug:
+       msg: We are doing somethin
+     when: result.objects | length == 1
+
+Of course, you can also loop over the result using a *loop* construct:
+
+.. code-block:: yaml
+
+   - name: Fetch list of all assets
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
   - name: Fetch a list of all assets
```

File: docs/source/modules.rst [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+       builds: ...
+
+Almost every module for manipulating resources has its info counterpart that
```
Comment (mancabizjak) [inline docs/source/modules.rst]: I would rephrase this a little, something along the lines: "Almost every module for manipulating resources has its counterpart module that can be used to retrieve information about the corresponding resources, for instance _asset_ and _asset_info_ modules".

I think that "has its info counterpart" is a bit confusing for someone not familiar with the collection.

File: docs/source/modules.rst [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+can be used to retrieve information about the certain resources. For example,
+to get a list of assets from the ``my-namespace`` namespace, we would write a
+task like this one:
+
+.. code-block:: yaml
+
+   - name: Fetch list of all assets
+     asset:
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
     asset_info:
```

File: docs/source/modules.rst [hunk 16]
```diff
@@ -0,0 +1,146 @@
...
+       namespace: my-namespace
+     register: result
+
+We can also retrieve information about a single asset by adding a *name*
+parameter to the previous task:
+
+.. code-block:: yaml
+
+   - name: Fetch a specific asset
+     asset:
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
     asset_info:
```

File: docs/source/modules.rst [hunk 17]
```diff
@@ -0,0 +1,146 @@
...
+
+   - name: Fetch list of all assets
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+     register: result
+
```
Comment (mancabizjak) [inline docs/source/modules.rst]: Here, I would add a note "Note the usage of the _asset_info_ module in the example above".

File: docs/source/modules.rst [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+Info modules always return a list of objects that Sensu API returned. And if
+we try to fetch a non-existing resource, the result will hold an empty list.
+This makes it easy to write a conditional tasks using next pattern:
+
+.. code-block:: yaml
+
+   - name: Fetch a specific asset
+     asset:
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
     asset_info:
```

File: docs/source/modules.rst [hunk 19]
```diff
@@ -0,0 +1,146 @@
...
+     debug:
+       msg: We are doing somethin
+     when: result.objects | length == 1
+
+Of course, you can also loop over the result using a *loop* construct:
+
+.. code-block:: yaml
+
+   - name: Fetch list of all assets
+     asset:
```
Comment (mancabizjak) [inline docs/source/modules.rst]:
```suggestion
     asset_info:
```

File: docs/source/modules.rst [hunk 20]
```diff
@@ -0,0 +1,146 @@
...
+
+   - name: Make sure asset is absent
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+       state: absent
```
Comment (mancabizjak) [inline docs/source/modules.rst]: Can we move this part right after the example that demonstrates creation of an asset? I think it's easier to follow along if we first show a couple of tasks that manipulate resources using _asset_, and then a couple of tasks just fetching info via the _asset_info_ module. Otherwise one gets confused easily.

File: docs/source/quickstart.rst [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+   :language: yaml
+
+When we run it, Ansible will install and configure backend and agents on
+selected hosts, and then configure a ntp check that agents will execute twice
+a minute. Note that we do not need to inform agents explicitly where the
+backend is because the :doc:`agent role <roles/agent>` can obtain the
+backend's address from the inventory.
+
+Now, before we can run this playbook, we need to prepare an inventory file.
+The inventory should contain two groups of hosts: backends and agents. A
```
Comment (mancabizjak) [inline docs/source/quickstart.rst]:
```suggestion
The inventory should contain two groups of hosts: _backends_ and _agents_. A
```

File: docs/source/quickstart.rst [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+   :language: yaml
+
+Replace the IP addresses with your own and make sure you can ssh into those
+hosts. If you need help with building your inventory file, consult `official
+documentation on inventory`_.
+
+.. _official documentation on inventory:
+   https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
+
+All that we need to do now is to install the `sensu.sensu_go` collection and
```
Comment (mancabizjak) [inline docs/source/quickstart.rst]:
```suggestion
All that we need to do now is to
```
We already installed it in the first step ;)

File: docs/examples/quickstart/playbook.yaml [hunk 23]
```diff
@@ -0,0 +1,50 @@
+---
+- name: Install backend
```
Comment (mancabizjak) [inline docs/examples/quickstart/playbook.yaml]:
```suggestion
- name: Install, configure and run Sensu backend
```

File: docs/examples/quickstart/playbook.yaml [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+  hosts: backends
+  become: true
+
+  tasks:
+    - name: Install backend
+      include_role:
+        name: sensu.sensu_go.backend
+
+- name: Install agents
```
Comment (mancabizjak) [inline docs/examples/quickstart/playbook.yaml]:
```suggestion
- name: Install, configure and run Sensu agents
```

**Comment by mancabizjak** [comment]
I just remembered another thing: I noticed that all statements mentioning the Ansible 2.9 requirement were removed here. Is this on purpose? I'm thinking that since collections are a relatively new gig, including a brief note about this somewhere in the docs is not completely out of context :)

File: docs/source/installation.rst [hunk 25]
```diff
@@ -0,0 +1,80 @@
...
+
+This last method of installation might come in handy in situations where our
+Ansible control node cannot access Ansible Galaxy or Automation Hub.
+
+First, we need to download the Sensu Go Ansible collection archive from the
+GitHub `releases page`_ and then transfer that archive to the Ansible control
+node. Once we have that archive on our control node, we can install the Sensu
+Go collection by running::
+
+   $ ansible-galaxy collection install path/to/sensu-sensu_go-1.0.0.tar.gz
```
[the remark this answers is not in the document]
Comment (tadeboro) [reply docs/source/installation.rst]: Tarballs that GitHub prepares automatically for each release do not contain enough information for the `ansible-galaxy` tool. We will need to upload a separate build artifact for each release and attach it to certain release. I described this process in the WIP documentation for developers, but I will also add a note for end-users about what file they should download. Thanks for catching things like this!

File: docs/source/roles/agent.rst [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+.. _official Sensu documentation:
+   https://docs.sensu.io/sensu-go/latest/reference/agent/#configuration
+
+.. note::
+
+   Role copies the key-value pairs from the *agent_config* variable verbatim
+   to the configuration file. This means that we must copy the key names
+   **exactly** as they appear in the configuration reference. In a way, the
+   *agent_config* variable should contain a properly indented copy of the
+   ``/etc/sensu/backend.yml`` file.
```
Comment (mancabizjak) [inline docs/source/roles/agent.rst]:
```suggestion
   ``/etc/sensu/agent.yml`` file.
```

File: docs/source/modules.rst (hunk 19 above)
Comment (mancabizjak) [reply docs/source/modules.rst]: We forgot this one ;)
